From cedcf7d9d3c6a55c2aaa9da447073124e83d4b2d Mon Sep 17 00:00:00 2001
From: James Sadler <freshtonic@gmail.com>
Date: Tue, 27 Apr 2010 18:54:57 +1000
Subject: [PATCH] Fixed use of Ruby 1.8 incompatible syntax.

This fixes a test suite failure when running the tests under 1.8.
---
 activesupport/lib/active_support/json/encoding.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/activesupport/lib/active_support/json/encoding.rb b/activesupport/lib/active_support/json/encoding.rb
index 0f38fd0..e692f6d 100644
--- a/activesupport/lib/active_support/json/encoding.rb
+++ b/activesupport/lib/active_support/json/encoding.rb
@@ -104,7 +104,7 @@ module ActiveSupport
 
         def escape(string)
           if string.respond_to?(:force_encoding)
-            string = string.encode(::Encoding::UTF_8, undef: :replace).force_encoding(::Encoding::BINARY)
+            string = string.encode(::Encoding::UTF_8, :undef => :replace).force_encoding(::Encoding::BINARY)
           end
           json = string.
             gsub(escape_regex) { |s| ESCAPED_CHARS[s] }.
-- 
1.6.6.1

